packages/features/eventtypes/lib/getPublicEvent.ts

```diff
@@ -244,7 +244,12 @@ export const getPublicEvent = async (
     hosts: hosts,
   };
 
-  const users = getUsersFromEvent(eventWithUserProfiles) || (await getOwnerFromUsersArray(prisma, event.id));
+  let users = getUsersFromEvent(eventWithUserProfiles) || (await getOwnerFromUsersArray(prisma, event.id));
```

Comment:
@Udit-takkar We already call `getOwnerFromUsersArray(prisma, event.id)` here in the OR and then you added it again down below which seems odd. 

Why do we need this call twice?

Reply:
getUsersFromEvent was returning an empty array in case of legacy team event types where hosts[] is empty but users[] existed because of 

```
  if (team) {
    return (hosts || []).filter((host) => host.user.username).map(mapHostsToUsers);
  }

```

and getOwnerFromUsersArray(prisma, event.id) wouldn't be executed if empty array is returned.